lib/dao.js

```diff
@@ -457,12 +467,27 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
 
         var connector = self.getConnector();
 
-        if (Model.settings.validateUpsert === false) {
+        var doValidate = undefined;
+        if (options.validate === undefined) {
+          if (Model.settings.validateUpsert === undefined) {
+            if (Model.settings.automaticValidation === undefined) {
+
+            } else {
+              doValidate = Model.settings.automaticValidation;
+            }
+          } else {
+            doValidate = Model.settings.validateUpsert
+          }
+        } else {
+          doValidate = options.validate;
+        }
+
+        if (doValidate === false) {
           callConnector();
         } else {
           inst.isValid(function(valid) {
             if (!valid) {
-              if (Model.settings.validateUpsert) {
+              if (doValidate) { //backwards compatibility with validateUpsert:undefined
```

Comment:
> I also use a local variable doValidate instead of setting options.validate, because some test required the options to stay the same during the lifecycle. I did that in save, but the tests completed successfully even when changing the options. Should it be changed there as well?

It's better to treat input arguments as immutable and use a local variable (e.g. `doValidate`) everywhere.